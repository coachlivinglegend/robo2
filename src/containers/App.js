import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { fetchRobots } from '../state/features/robots';

import './App.css';
import { Route, Routes } from 'react-router-dom';
import Home from '../pages/Home';

const App = () => {
  const searchFieldRedux = useSelector((state) => state.searchField.value);
  const robots = useSelector((state) => state.robots.value);
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(fetchRobots());
  }, [dispatch]);

  const filteredRobots = robots.filter((rob) =>
    rob.name.toLocaleLowerCase().includes(searchFieldRedux)
  );

  if (robots.length === 0) {
    return <h1>Loading</h1>;
  }

  return (
    <Routes>
      <Route path='/*' element={<Home filteredRobots={filteredRobots} />} />
      <Route path='/else' element={<h1>dunno</h1>} />
    </Routes>
  );
};

export default App;
