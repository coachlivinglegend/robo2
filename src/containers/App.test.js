import { shallow, render } from 'enzyme';
import { Provider } from 'react-redux';
import App from './App';
import configureStore from 'redux-mock-store';
import { Router } from 'react-router-dom';

const middlewares = [];
const mockStore = configureStore(middlewares);
const robots = [
  {
    id: 1,
    name: 'John Doe',
    username: 'joneddidoe',
    email: 'sssskkkkkkkkkrrrrrrrrr@io',
  },
  {
    id: 2,
    name: 'John Doe',
    username: 'joneddidoe',
    email: 'sssskkkkkkkkkrrrrrrrrr@io',
  },
];

const initialState = {
  robots: { value: [], status: null },
  searchField: { value: '' },
};
test('expect to take a snapshot', () => {
  const store = mockStore(initialState);
  expect(
    render(
      <Provider store={store}>
        <App />
      </Provider>
    )
  ).toMatchSnapshot();
});
