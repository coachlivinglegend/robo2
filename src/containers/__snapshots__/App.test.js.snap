// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to take a snapshot 1`] = `
<h1>
  Loading
</h1>
`;
